atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraRequestExceptionHandler.java

```diff
@@ -143,13 +144,14 @@ boolean shouldBlacklist(Exception ex, int numberOfAttempts) {
             return;
         }
 
+        long backOffPeriod = strategy.getBackoffPeriod(req.getNumberOfAttemptsOnHost(hostTried));
         log.info("Retrying a query, {}, with backoff of {}ms, intended for host {}.",
                 UnsafeArg.of("queryString", req.getFunction().toString()),
-                SafeArg.of("sleepDuration", strategy.getBackoffPeriod(req.getNumberOfAttempts())),
+                SafeArg.of("sleepDuration", backOffPeriod),
                 SafeArg.of("hostName", CassandraLogHelper.host(hostTried)));
 
         try {
-            Thread.sleep(strategy.getBackoffPeriod(req.getNumberOfAttempts()));
+            Thread.sleep(backOffPeriod);
         } catch (InterruptedException i) {
             throw new RuntimeException(i);
```

Comment:
sorry about this - while you're here, please reinterrupt the thread (`Thread.currentThread().interrupt();`) - the original code was a bit buggy. See [article](https://www.ibm.com/developerworks/library/j-jtp05236/index.html)